File: framework/languages/dotnet/samples/TicTacToe.SessionGateway/Server/Api/CreateMatchHandler.cs
using TicTacToe.SessionGateway.Shared.Configuration;
using TicTacToe.SessionGateway.Shared.Contracts;
using Zlink.Framework.Contracts.Channels;
using Zlink.Framework.Contracts.Handlers;

namespace TicTacToe.SessionGateway.Api;

[ZLinkHandlerGroup("api")]
internal sealed class CreateMatchHandler(IZLinkClientServerClient client)
{
    [ZLinkRequest]
    public async ValueTask<CreateMatchRes> CreateMatch(
        CreateMatchReq request,
        ZLinkRequestContext context,
        CancellationToken cancellationToken)
    {
        _ = context;
        var ownerActorId = request.OwnerActorId?.Trim();
        if (string.IsNullOrWhiteSpace(ownerActorId))
        {
            throw new InvalidOperationException("Match owner actor id must not be empty.");
        }

        var room = await client.Request(
                SampleNames.PlayChannel,
                new CreateMatchRoomReq())
            .Timeout(SampleTimings.RequestTimeout)
            .SubmitAsync<CreateMatchRoomRes>(cancellationToken)
            .ConfigureAwait(false);

        return new CreateMatchRes(room.MatchId, ownerActorId);
    }
}

File: framework/languages/node/samples/TicTacToe.SessionGateway/Server/Api/Handlers/create-match-handler.js
const { Inject } = require('@nestjs/common');
const { PacketNames } = require('../../../Shared/Contracts/messages');
const { SampleNames, SampleTimings } = require('../../../Shared/Configuration/sample-names');
const { PLAY_ROUTE_CLIENT } = require('../api-tokens');

class CreateMatchHandler {
  constructor(playClient) {
    this.playClient = playClient;
  }

  async handle(request) {
    return await this.playClient.request(
      SampleNames.playNode,
      PacketNames.createMatchReq,
      { matchName: request.matchName ?? 'match' },
      SampleTimings.requestTimeout
    );
  }
}

Inject(PLAY_ROUTE_CLIENT)(CreateMatchHandler, undefined, 0);

module.exports = { CreateMatchHandler };
